provider/core/src/hello_world.rs

```diff
@@ -98,9 +100,10 @@ impl HelloWorldProvider {
         ("zh", "你好世界"),
     ];
 
-    /// Converts this provider into one that serves JSON blobs of the same data.
+    /// Converts this provider into a [`BufferProvider`] that uses JSON serialization.
+    #[cfg(feature = "deserialize_json")]
     pub fn into_json_provider(self) -> HelloWorldJsonProvider {
```

Comment:
Thought: Putting a default-feature function behind a non-default feature is technically a breaking change

Reply:
Yeah...I think it's fine because unless you want to look at raw bytes, it's not usable without the feature anyway, and I'd be surprised if there was actually any client code written against this API